packages/schematics/angular/utility/validation.ts

```diff
@@ -10,7 +10,8 @@ import { SchematicsException } from '@angular-devkit/schematics';
 
 // Must start with a letter, and must contain only alphanumeric characters or dashes.
 // When adding a dash the segment after the dash must also start with a letter.
-export const htmlSelectorRe = /^[a-zA-Z][.0-9a-zA-Z]*(:?-[a-zA-Z][.0-9a-zA-Z]*)*$/;
+export const htmlSelectorRe =
+  /^[a-zA-Z][.0-9a-zA-Z]*((:?-[0-9]+)*|(:?-[a-zA-Z][.0-9a-zA-Z]*(:?-[0-9]+)*)*)$/;
```

Comment:
Regexp used [here](https://github.com/angular/angular-cli/blob/b66d94e5949102b2235b89cb75866973986ea30a/packages/angular_devkit/schematics/src/formats/html-selector.ts#L17C1-L44) doesn't validate html selector correctly. For eg., it considers `foo-` as a valid selector which is wrong, right? And it also allows underscore. 

I have modified regexp in validation.ts file itself to cover selectors like `one-1` but if we still want to use the logic at `html-selector.ts` then we might have to make some changes in that logic.